contentprovider.go

```diff
@@ -552,6 +552,48 @@ func findSection(secs []DocumentSection, off, sz uint32) (uint32, bool) {
 	return 0, false
 }
 
+// findMaxOverlappingSection returns the index of the section in secs that
+// overlaps the most with the area defined by off and sz, relative to the size
+// of the section. If no section overlaps, it returns 0, false. If multiple
+// sections overlap the same amount, the first one is returned.
+//
+// The implementation assumes that sections do not overlap and are sorted by
+// DocumentSection.Start.
+func findMaxOverlappingSection(secs []DocumentSection, off, sz uint32) (uint32, bool) {
+	// Find the first section that might overlap
+	j := sort.Search(len(secs), func(i int) bool { return secs[i].End > off })
+
+	if j == len(secs) || secs[j].Start >= off+sz {
```

Comment:
minor: I don't know if this makes the code better, but I felt like I had to keep ensuring that .Start was with off and .End was off+sz. Maybe two local variables `start := off; end := off+sz` would make it more obvious at each site we do a comparison that we are doing the right one?